source/opt/local_single_store_elim_pass.cpp

```diff
@@ -174,7 +174,8 @@ bool LocalSingleStoreElimPass::RewriteDebugDeclares(Instruction* store_inst,
         context()->GetDominatorAnalysis(store_block->GetParent());
     for (auto* decl : invisible_decls) {
       if (dominator_analysis->Dominates(store_inst, decl)) {
-        context()->get_debug_info_mgr()->AddDebugValueForDecl(decl, value_id);
+        context()->get_debug_info_mgr()->AddDebugValueForDecl(decl, value_id,
+                                                              decl);
```

Comment:
Just an observation: in most usages, the first and last parameter passed to `AddDebugValueForDecl` is the same. I'm not familiar enough with the it to know whether there's an opportunity for improvement.

Reply:
Yes, but there are two exceptions: [DebugInfoManager::AddDebugValueIfVarDeclIsVisible()](https://github.com/KhronosGroup/SPIRV-Tools/pull/4022/files#diff-6e5ffb311b0364a87e9794b09d9efa5d5a431fa651b137dce6b1715c8bc9be53R503) and [ScalarReplacementPass::ReplaceWholeDebugDeclare()](https://github.com/KhronosGroup/SPIRV-Tools/pull/4022/files#diff-abc97f3e780d673299d2def447ad569d267b35f9a476f189a851b83317b6adb0R178).

In my opinion, it would be better to keep the third parameter.